erts/emulator/beam/jit/x86/beam_asm.hpp

```diff
@@ -709,6 +723,9 @@ class BeamAssembler : public ErrorHandler {
         a.sub(x86::rsp, imm(15));
         a.and_(x86::rsp, imm(-16));
 #endif
+        if (hasCpuFeature(CpuFeatures::X86::kAVX)) {
+            a.vzeroupper();
+        }
```

Comment:
```suggestion
        /* If the emulator has not been compiled with AVX support (which stops it
         * from using legacy SSE instructions), we'll need to clear the upper bits of
         * all AVX registers to avoid AVX/SSE transition penalties.  */
#if !defined(__AVX__)
        if (hasCpuFeature(CpuFeatures::X86::kAVX)) {
            a.vzeroupper();
        }
#endif
```